File: 백준/Silver/1620. 나는야 포켓몬 마스터 이다솜/나는야 포켓몬 마스터 이다솜.py
import sys

N, M = [int(x) for x in input().split()]
number_pokemon_dictionary = {}
pokemon_number_dictionary = {}

for i in range(1, N + 1):
    pokemon = sys.stdin.readline()[:-1]
    number_pokemon_dictionary[i] = pokemon
    pokemon_number_dictionary[pokemon] = i


def return_result(text) -> str:
    if text[0].isdigit():
        return number_pokemon_dictionary[int(text[:-1])]
    else:
        return str(pokemon_number_dictionary[text[:-1]])


for _ in range(M):
    print(return_result(sys.stdin.readline()))

File: 백준/Silver/1620. 나는야 포켓몬 마스터 이다솜/나는야 포켓몬 마스터 이다솜.swift
import Foundation

let nm = readLine()!.split(separator:" ").compactMap { Int($0) },
    n = nm[0],
    m = nm[1]
var numberPokemonDictionary = [Int : String](),
    pokemonNumberDictionary = [String : Int]()
    
for i in (1..<(n+1)) {
    let pokemon = readLine()!
    numberPokemonDictionary[i] = pokemon
    pokemonNumberDictionary[pokemon] = i
}

func search(_ text: String) -> String {
    if text.first!.isNumber, let number = Int(text) {
        return numberPokemonDictionary[number, default: ""]
    } else {
       return String(pokemonNumberDictionary[text, default:1]) 
    }
}

for _ in (0..<m) {
    print(search(readLine()!))
}